src/pages/proposals/proposal-details/details.tsx

```diff
@@ -0,0 +1,200 @@
+import { observer } from 'mobx-react-lite';
+import React, { FunctionComponent } from 'react';
+import { useStore } from 'src/stores';
+import { useParams } from 'react-router-dom';
+import { Loader } from 'src/components/common/loader';
+import { VictoryContainer, VictoryPie, VictoryTooltip } from 'victory';
+import { useVotingDetails } from 'src/hooks/use-voting-details';
+import moment from 'moment';
+
+interface ProposalDetailParam {
+	proposalId: string;
+}
+
+const URL_REGEX = /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)/;
+
+const ProposalDetailPage: FunctionComponent = observer(() => {
+	const { proposalId } = useParams<ProposalDetailParam>();
+	const { voteCalculation } = useVotingDetails();
+	const { chainStore, queriesStore } = useStore();
+	const queries = queriesStore.get(chainStore.current.chainId);
+	const proposal = queries.cosmos.queryGovernance.getProposal(proposalId);
+
+	const statusColor = (status: number) => {
+		switch (status) {
+			case 2:
+				return 'text-enabledGold';
+			case 3:
+				return 'text-pass';
+			case 4:
+				return 'text-missionError';
+
+			default:
+				return 'text-primary-50';
+		}
+	};
+
+	const formatLinks = (txt: string) => {
```

Comment:
All this functions can be moved outside of the component render function, we don't need to declare them every time the component renders